pandas/core/common.py

```diff
@@ -2439,6 +2439,8 @@ def _get_dtype_type(arr_or_dtype):
         return np.dtype(arr_or_dtype).type
     elif isinstance(arr_or_dtype, CategoricalDtype):
         return CategoricalDtypeType
+    elif arr_or_dtype is NotImplemented:
+        return type(NotImplemented)
```

Comment:
This function just returns the type of an object ... NotImplemented is an object. So not sure what the question is?

if the question is why is this feature important? Pandas has code that checks the type of return values at acts accordingly. Occasionally Numpy will return this type and this code breaks with an attribute error because NotImplementedType has no attribute dtype.